components/html-2-objects-compare.js

```diff
@@ -0,0 +1,175 @@
+            description: "path to the html file to be created.",
+            addressable: true,
+            required: true
+        },
+        left: { 
+            datatype: 'object',
+            description: "a JavaScript object to display on the left side side of the HTML page.",
+            addressable: true,
+            required: true
+        },
+        right: { 
+            datatype: 'object',
+            description: "a JavaScript object to display on the right side side of the HTML page.",
+            addressable: true,
+            required: true
+        }
+    },
+
+    /**
+     * This updater will generate a simple HTML page that visually displays the content of 
+     * both the left and right objects.
+     *
+     * @param file file path to which the html should be written
+     * @param left Javascript object to display on left side of HTML page and an IIP 
+     *             setting of format {"title":"<title>"}
+     * @param right Javascript object to display on right side of HTML page and an IIP 
+     *             setting of format {"title":"<title>"}
+     */
+    updater: function( file, left, right ) {   
+
+        // Verify that we have all data required to generate the html page. 
+        // TODO: Revisit and remove this check after more of the pipeline architecture is completed.
+        if ( shouldFireUpdater( file, left, right) ) { 
+
+            // We have both an IIP and port data on these ports.  It can theoretically arrive in any order.
+            // sortPortData return an object with a title and a data field so we know what we are referencing
+            // henceforth.
+            var leftOutput = sortPortData( left, 'left' );
+            var rightOutput = sortPortData( right, 'right' );
+
+            var content = "<html>\n" +
+                          "  <head>\n" +
+                          "    <title> Comparison of "+leftOutput.title+" &amp; "+rightOutput.title+" Data </title>\n" +
+                          "  </head>\n" +
+                          "  <body>\n" +
+                          "      <table cellpadding='10' width='100%'>\n" +
+                          "         <tr>\n" + 
+                          "           <th bgcolor='#c8c8c8' width='50%'> " + leftOutput.title + " </th>" + 
+                          "           <th bgcolor='#c8c8c8' width='50%'> " + rightOutput.title + " </th>" + 
+                          "         </tr>\n" +
+                          "         <tr>\n" + 
+                          "           <td bgcolor='#e8e8e8'> <pre> "+ util.inspect( leftOutput.data, {depth: null} ) + "</pre></td>" +
+                          "           <td> <pre> "+ util.inspect( rightOutput.data, {depth: null} ) + "</pre></td>";
+                          "         </tr>\n" +
+                          "      </table>\n" +
+                          "  </body>\n" +
+                          "</html>";
+            fs.writeFileSync(file, content);
```

Comment:
In the spirit of doing only one thing, what about just producing the content and letting a down stream component determine what to do with it?